src/resolve/resolved-dns-rr.c

```diff
@@ -321,7 +321,7 @@ char* dns_resource_key_to_string(const DnsResourceKey *key, char *buf, size_t bu
         c = dns_class_to_string(key->class);
         t = dns_type_to_string(key->type);
 
-        snprintf(buf, buf_size, "%s %s%s%.0u %s%s%.0u",
+        snprintf(buf, buf_size, "%s %s%s%.0d %s%s%.0d",
```

Comment:
PRIu16

Reply:
```FAILED: src/resolve/085a0f9@@systemd-resolve-core@sta/resolved-dns-rr.c.o 
cc -Isrc/resolve/085a0f9@@systemd-resolve-core@sta -Isrc/resolve -I../src/resolve -Isrc/basic -I../src/basic -Isrc/boot -I../src/boot -Isrc/home -I../src/home -Isrc/shared -I../src/shared -Isrc/systemd -I../src/systemd -Isrc/journal -I../src/journal -Isrc/journal-remote -I../src/journal-remote -Isrc/nspawn -I../src/nspawn -Isrc/timesync -I../src/timesync -I../src/time-wait-sync -Isrc/login -I../src/login -Isrc/udev -I../src/udev -Isrc/libudev -I../src/libudev -Isrc/core -I../src/core -Isrc/shutdown -I../src/shutdown -I../src/libsystemd/sd-bus -I../src/libsystemd/sd-device -I../src/libsystemd/sd-event -I../src/libsystemd/sd-hwdb -I../src/libsystemd/sd-id128 -I../src/libsystemd/sd-netlink -I../src/libsystemd/sd-network -I../src/libsystemd/sd-resolve -Isrc/libsystemd-network -I../src/libsystemd-network -I. -I.. -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=gnu99 -g -Wno-unused-parameter -Wno-missing-field-initializers -Wno-unused-result -Wno-format-signedness -Werror=undef -Wlogical-op -Wmissing-include-dirs -Wold-style-definition -Wpointer-arith -Winit-self -Wfloat-equal -Wsuggest-attribute=noreturn -Werror=missing-prototypes -Werror=implicit-function-declaration -Werror=missing-declarations -Werror=return-type -Werror=incompatible-pointer-types -Werror=format=2 -Wstrict-prototypes -Wredundant-decls -Wmissing-noreturn -Wimplicit-fallthrough=5 -Wshadow -Wendif-labels -Wstrict-aliasing=2 -Wwrite-strings -Werror=overflow -Werror=shift-count-overflow -Werror=shift-overflow=2 -Wdate-time -Wnested-externs -Wno-maybe-uninitialized -ffast-math -fno-common -fdiagnostics-show-option -fno-strict-aliasing -fvisibility=hidden -fstack-protector -fstack-protector-strong --param=ssp-buffer-size=4 -Werror=shadow -include config.h -fPIC -MD -MQ 'src/resolve/085a0f9@@systemd-resolve-core@sta/resolved-dns-rr.c.o' -MF 'src/resolve/085a0f9@@systemd-resolve-core@sta/resolved-dns-rr.c.o.d' -o 'src/resolve/085a0f9@@systemd-resolve-core@sta/resolved-dns-rr.c.o' -c ../src/resolve/resolved-dns-rr.c
../src/resolve/resolved-dns-rr.c: In function ‘dns_resource_key_to_string’:
../src/resolve/resolved-dns-rr.c:324:44: error: unknown conversion type character ‘P’ in format [-Werror=format=]
  324 |         snprintf(buf, buf_size, "%s %s%s%.0PRIu16 %s%s%.0PRIu16",
      |                                            ^
../src/resolve/resolved-dns-rr.c:324:52: error: format ‘%s’ expects argument of type ‘char *’, but argument 7 has type ‘int’ [-Werror=format=]
  324 |         snprintf(buf, buf_size, "%s %s%s%.0PRIu16 %s%s%.0PRIu16",
      |                                                   ~^
      |                                                    |
      |                                                    char *
      |                                                   %d
  325 |                  dns_resource_key_name(key),
  326 |                  strempty(c), c ? "" : "CLASS", c ? 0 : key->class,
      |                                                 ~~~~~~~~~~~~~~~~~~
      |                                                       |
      |                                                       int
../src/resolve/resolved-dns-rr.c:324:58: error: unknown conversion type character ‘P’ in format [-Werror=format=]
  324 |         snprintf(buf, buf_size, "%s %s%s%.0PRIu16 %s%s%.0PRIu16",
      |                                                          ^
../src/resolve/resolved-dns-rr.c:324:33: error: too many arguments for format [-Werror=format-extra-args]
  324 |         snprintf(buf, buf_size, "%s %s%s%.0PRIu16 %s%s%.0PRIu16",
      |                                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~```

Reply:
`PRIu16` is a format string, so it should be used as:

```c
snprintf(buf, buf_size, "%s %s%s%.0d %s%s%" PRIu16,
...
```

Reply:
See the examples here:

https://en.cppreference.com/w/c/types/integer

Reply:
sorry, fixed locally, just trying to get https://github.com/systemd/systemd/pull/15729#issuecomment-624905375 patched but having some issues

Reply:
fixed

Reply:
hmm, not actually fixed, is it?